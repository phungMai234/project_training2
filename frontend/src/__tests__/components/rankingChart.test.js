import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import RankingChart from "../../components/ranking/rankingChart";

const mockStore = configureStore([]);

describe('Ranking Chart Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      query: {
        data: {
          startDate: "2019/04/29",
          endDate: "2019/04/29"
        }
      },
      rankings: {
        data: [20, 4, 5, 7, 2, 4, 8],
        isFetching: false
      }
    });
    component = renderer.create(
      <Provider store={store}>
        <RankingChart/>
      </Provider>
    );
  })
  it('should render Ranking Chart with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
})