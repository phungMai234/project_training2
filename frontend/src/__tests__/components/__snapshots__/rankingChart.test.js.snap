// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ranking Chart Component should render Ranking Chart with given state from Redux store 1`] = `
<div
  className="ranking"
>
  <label
    className="text-secondary"
  >
    Ranking
  </label>
  <div
    className="d-flex justify-content-between"
  >
    <span>
      Day
    </span>
    <span>
      Number
    </span>
  </div>
  <div
    className="progress d-flex justify-content-between"
  >
    <div
      className="progress-bar align-items-start"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <span>
        Day 
        1
      </span>
    </div>
    <span>
      20
    </span>
  </div>
  <div
    className="progress d-flex justify-content-between"
  >
    <div
      className="progress-bar align-items-start"
      style={
        Object {
          "width": "20%",
        }
      }
    >
      <span>
        Day 
        2
      </span>
    </div>
    <span>
      4
    </span>
  </div>
  <div
    className="progress d-flex justify-content-between"
  >
    <div
      className="progress-bar align-items-start"
      style={
        Object {
          "width": "25%",
        }
      }
    >
      <span>
        Day 
        3
      </span>
    </div>
    <span>
      5
    </span>
  </div>
  <div
    className="progress d-flex justify-content-between"
  >
    <div
      className="progress-bar align-items-start"
      style={
        Object {
          "width": "35%",
        }
      }
    >
      <span>
        Day 
        4
      </span>
    </div>
    <span>
      7
    </span>
  </div>
  <div
    className="progress d-flex justify-content-between"
  >
    <div
      className="progress-bar align-items-start"
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <span>
        Day 
        5
      </span>
    </div>
    <span>
      2
    </span>
  </div>
  <div
    className="progress d-flex justify-content-between"
  >
    <div
      className="progress-bar align-items-start"
      style={
        Object {
          "width": "20%",
        }
      }
    >
      <span>
        Day 
        6
      </span>
    </div>
    <span>
      4
    </span>
  </div>
  <div
    className="progress d-flex justify-content-between"
  >
    <div
      className="progress-bar align-items-start"
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <span>
        Day 
        7
      </span>
    </div>
    <span>
      8
    </span>
  </div>
</div>
`;
